### PR TITLE
Inbox private message form

### LINKED ISSUE #748: When your inbox is empty, it should help you figure out how to send messages.
Maybe have a sentence explaining how to send a message, or what messages are. Or some feature to find people to message....?

inbox

### PR DESCRIPTION
**Pull request by aweiksnar**
- Adds a form to send new private messages to the inbox page
- Closes #748
- Using user search for selection, so related to https://github.com/zooniverse/Panoptes/issues/1034

### CODE REVIEW

**Comment by aweiksnar** [comment]
If you think it's worth waiting until user search is more fleshed out I'm cool with that, but figured I'd put this out there.

**Comment by chrissnyder** [comment]
Might be. Playing with it a bit now and it's difficult to separate out the two problems. Will look more after lunch.

File: app/talk/inbox-form.cjsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    subjectErrors = getErrors(subject, subjectValidations)
+    messageValidationErrors = getErrors(message, messageValidations)
+
+    validationErrors = userErrors.concat subjectErrors.concat messageValidationErrors
+    @setState {validationErrors}
+    !!validationErrors.length
+
+  onSubmitMessage: (e) ->
+    e.preventDefault()
+    form = findDOMNode(@).querySelector('form')
```
Comment (chrissnyder) [inline app/talk/inbox-form.cjsx]: Any reason you use this style over refs?

File: app/talk/inbox-form.cjsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    recipient_ids = form.querySelector('[name="userids"]')
+      .value.split(',').map (id) -> parseInt(id)
+      .filter(Number)
+
```
Comment (chrissnyder) [inline app/talk/inbox-form.cjsx]: Given that we are limiting recipients to one person, couldn't this just be `form.querySelector('[name="userids"]').value`?

File: app/talk/inbox-form.cjsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    title = findDOMNode(@refs.subject).value
+    body = form.querySelector('textarea').value
+    user_id = @props.user.id
+
+    conversation = {title, body, user_id, recipient_ids}
+
+    errors = @validations(body, title, recipient_ids)
+    return errors if errors.length
```
Comment (chrissnyder) [inline app/talk/inbox-form.cjsx]: This should just be `return errors if errors`

**Comment by chrissnyder** [comment]
Separating out the validations is pretty nice. For future plans here, perhaps a `<ValidatedInput>` component would be useful so each form field handles it's own validations?

File: app/talk/inbox-form.cjsx (hunk 2 above)
Comment (aweiksnar) [inline app/talk/inbox-form.cjsx]: Ah yeah (but it'd have to be an array). There could be multiple recipients, but I wanted to limit to one until I get to #738

Comment (chrissnyder) [inline app/talk/inbox-form.cjsx]: My bad then, didn't realize a message could have multiple recipients.

File: app/talk/inbox-form.cjsx (hunk 1 above)
Comment (aweiksnar) [inline app/talk/inbox-form.cjsx]: You're right, refs would be cleaner here. I'll update in a sec

File: app/talk/inbox-form.cjsx (hunk 3 above)
Comment (aweiksnar) [inline app/talk/inbox-form.cjsx]: Good call, I originally had validations return an array but forgot to update

**Comment by aweiksnar** [comment]
Good idea on that validated input component, I'll look to add that soon. I just updated the commit and think I hit all of your points

**Comment by chrissnyder** [comment]
:+1:

**Pull request merged**